sdk/src/main/java/com/uid2/UID2Manager.kt

```diff
@@ -177,6 +166,35 @@ public class UID2Manager internal constructor(
             checkIdentityRefresh()
         }
 
+    /**
+     * Adds a listener which can be used to determine if the [UID2Manager] instance has finished initializing.
+     * Initializing includes any time required to restore a previously persisted [UID2Identity] from storage.
+     *
+     * If a listener is added *after* initialization is complete, the callback will be invoked immediately.
+     */
+    public fun addOnInitializedListener(listener: () -> Unit): Unit = runBlocking {
+        initializedLock.withLock {
+            // If we've already finished initializing, we should immediately invoke the callback.
+            if (initialized.isCompleted) {
+                listener()
+            } else {
+                onInitializedListeners += listener
+            }
+        }
+    }
+
+    /**
+     * Removes a listener which was previously added via [addOnInitializedListener].
+     *
+     * If the listener has already been invoked after initialization was complete, it will already be removed internally
+     * to avoid leaking.
+     */
+    public fun removeOnInitializedListener(listener: () -> Unit): Unit = runBlocking {
+        initializedLock.withLock {
+            onInitializedListeners -= listener
+        }
+    }
```

Comment:
We can make these to be chainable at little to no cost.

```suggestion
    public fun addOnInitializedListener(listener: () -> Unit): UID2Manager = apply { 
      runBlocking {
        initializedLock.withLock {
            // If we've already finished initializing, we should immediately invoke the callback.
            if (initialized.isCompleted) {
                listener()
            } else {
                onInitializedListeners += listener
            }
        }
      }
    }

    /**
     * Removes a listener which was previously added via [addOnInitializedListener].
     *
     * If the listener has already been invoked after initialization was complete, it will already be removed internally
     * to avoid leaking.
     */
    public fun removeOnInitializedListener(listener: () -> Unit): UID2Manager = apply { 
      runBlocking {
        initializedLock.withLock {
            onInitializedListeners -= listener
        }
      }
    }
```